### PR TITLE
Add video snapshotting [#101046466]

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

**Comment by dougmartin** [comment]
@knowuh Could you review?

**Comment by scytacki** [comment]
Looks good to me. The key 'trick' is that canvas#drawImage can take a video element, and in that case it draws the current frame of the video.

File: app/scripts/shutterbug-worker.js [hunk 1]
```diff
@@ -99,10 +99,42 @@ ShutterbugWorker.prototype.getHtmlFragment = function(callback) {
...
+        $(elem).replaceWith(replacementCanvasImgs[i]);
+      });
+    }
+
+    // .addBack('video') handles case when the element itself is a video.
+    var replacementVideoImgs = [];
+    $element.find('video').addBack('video').map(function(i, elem) {
+      var $elem = $(elem);
+      var canvas = htmlTools.cloneDomItem($elem, "<canvas>");
+      canvas[0].getContext('2d').drawImage(elem, 0, 0, $elem.width(), $elem.height());
```
Comment (knowuh) [inline app/scripts/shutterbug-worker.js]: *Cool* – I didn't know that the drawImage method could take Dom elements as arguments.  I just looked it up in the [canvas drawImage api](https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/drawImage)

**Comment by knowuh** [comment]
Looks good.

**Pull request merged**